chargeCurrentPairsFinder.c

```diff
@@ -0,0 +1,88 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "chargeCurrentPairsFinder.h"
+
+int ascendingComparator(const void * a, const void * b) 
+{
+   return ( *(int*)a - *(int*)b );
+}
+
+static int validateCurrentSampleArray(int *chrgeCurrentArray, int size)
+{
+    if(size < 2)
+    {
+        return 0;
+    }
+
+    if(chrgeCurrentArray == NULL)
+    {
+        return 0;
+    }
+
+    return 1;
+}
+
+static int validateCurrentPair(int index, int firstElement, int lastElement)
```

Comment:
is it possible for index to have value as 0? in the case if index=0, then what is the significance of following if condition?

Reply:
index = 0 will be possible when the array size is 1. In that case, index shall be returned as 0. Similarly, first and last elements are same then I consider them as invalid pair. 

These are the 2 conditions where the return value of this function shall be returned 0.